core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -178,14 +178,21 @@ private[log] class LogCleanerManager(val logDirs: Seq[File],
           inProgress.contains(topicPartition) || isUncleanablePartition(log, topicPartition)
```

Comment:
Uncleanable partitions may come a result of IOException.  It is not always safe to obtain compaction lag for those uncleanable partitions since obtaining the firstbatchtime stamp might require some IOs.  We already have metrics for uncleanable partitions. I think it is safe not to account for uncleanable partitions in this case.

Reply:
uncleanable-partitions-count